### DISCUSSION TITLE
new_chat_members update not working

### DISCUSSION THREAD

**Discussion by jsjoeio** [telegraf/telegraf#1488] [Q&A]
I'm trying to use a special handler for `new_chat_members` but for some reason, when I add a new chat member to a group with my bot, it calls the `message` update type.

```typescript
const bot = new Telegraf(BOT_TOKEN)

bot.on("message", async (ctx) => {
  console.log("=======> this should only be called from new message")
  await handleOnMessage(ctx)
})

bot.on("new_chat_members", async (ctx) => {
  console.log("==============> new member added")
  const chatId = ctx.chat?.id
  if (chatId) {
    return await ctx.telegram.sendMessage(chatId, "hello new member")
  }
})
```

Any ideas what I'm doing wrong?

**Comment by jsjoeio** [comment]
Also, seems like the TypeScript types don't line up with the `message` that comes in when a new chat is added, specifically missing these two fields on `Message`:
```json
{
      "new_chat_participant":{
         "id":7365433292,
         "is_bot":false,
         "first_name":"Friend",
         "last_name":"redacted"
      },
      "new_chat_member":{
         "id":7365433292,
         "is_bot":false,
         "first_name":"Friend",
         "last_name":"redacted"
      },
}
```

**Reply by aahutsal** [reply]
Have you solved that, @jsjoeio ?

**Reply by jsjoeio** [reply]
Oh man...it's been a while. Let me check my code. 

Looks like I solved it by writing this helper function:

```typescript
export function isNewMemberUpdate(message: Message) {
  if (message.new_chat_members && message.new_chat_members.length > 0) {
    return true
  }
  return false
}
``` 

and then I have a general message handler:

```typescript
export async function handleOnMessage(ctx: TelegrafContext) {
  const { message } = ctx

  // Check if it's a new member joining a group
  if (typeof message !== "undefined" && isNewMemberUpdate(message)) {
    logger.info(`New member joined group: ${message}`)

    const replyMessage = NEW_MEMBER_REPLY(
      message.new_chat_members[0].first_name
    )

    await ctx.reply(replyMessage, {
      reply_to_message_id: message.message_id,
    })

    return
  }

bot.on("message", async (ctx) => {
  await handleOnMessage(ctx)
})
```

**Reply by aahutsal** [reply]
Well, I've hit similar issue yesterday.
The problem was:

When I had 
`bot.on('message'....`
then
`bot.on('new_chat_members'....`
It was not called

When I changed to have:
`bot.on('new_chat_members'....`
then
`bot.on('message'....`
and it worked.

Can you try that way? The custom handler is good, but it looks like a crutch :)

**Reply by jsjoeio** [reply]
This is the exact reason why I used `bot.on('message'...` because the other one was never called :/

**Comment by rankobp** [comment]
I have bot in 10k+ members chat added as admin. I can get all messages except when user is being added or removed from chat. Is this some Telegram limitation, can't find any info about this in the documentation?

Same code works just fine in small test group. I'm receiving both `bot.on(message('new_chat_members')` and `bot.on('message'`. In big group neither is being triggered when user enters or leaves group.

**Reply by MaGua-Bishop** [reply]
I had the same problem

**Reply by sammyajayi** [reply]
I am currently facing this same problem. found a way around it yet?

**Reply by MaGua-Bishop** [reply]
yes，I solved it

` bot.polling( allowed_updates=util.update_types)`

**Reply by MaGua-Bishop** [reply]
The same goes for webhook

`bot.set_webhook(f"xxxxx", allowed_updates=util.update_types)`

**Reply by MaGua-Bishop** [reply]
oh，I'm sorry,

This is a workaround for the pytelegrambotapi, but it works the same way

**Reply by alrf** [reply]
Any workarounds on it?
